overlord/hookstate/ctlcmd/list_connected.go

```diff
@@ -0,0 +1,91 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/overlord/ifacestate"
+)
+
+type listConnectedCommand struct {
+	baseCommand
+}
+
+var shortListConnectedHelp = i18n.G(`List names of connected plugs and slots`)
+var longListConnectedHelp = i18n.G(`
+The list-connected command lists the names of the connected plugs and slots this snap has.
+
+$ snapctl list-connected
+plug1
+slot2
+`)
+
+func init() {
+	addCommand("list-connected", shortListConnectedHelp, longListConnectedHelp, func() command {
+		return &listConnectedCommand{}
+	})
+}
+
+func (c *listConnectedCommand) Execute(args []string) error {
+	context, err := c.ensureContext()
+	if err != nil {
+		return err
+	}
+
+	snapName := context.InstanceName()
+
+	st := context.State()
+	st.Lock()
+	defer st.Unlock()
+
+	conns, err := ifacestate.ConnectionStates(st)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot get connections: %s", err)
+	}
+
+	var names []string
+	for refStr, connState := range conns {
+		if !connState.Active() {
+			continue
+		}
+		connRef, err := interfaces.ParseConnRef(refStr)
+		if err != nil {
+			return fmt.Errorf("internal error: %s", err)
+		}
+
+		if connRef.PlugRef.Snap == snapName {
+			names = append(names, connRef.PlugRef.Name)
+		}
+		if connRef.SlotRef.Snap == snapName {
+			names = append(names, connRef.SlotRef.Name)
+		}
+	}
+
+	sort.Strings(names)
+	for _, name := range names {
+		fmt.Fprintf(c.stdout, "%s\n", name)
```

Comment:
This could be simplified to:

```suggestion
		fmt.Fprintln(c.stdout, name)
```